app/src/main/java/com/azuresamples/msalnativeauthandroidkotlinsampleapp/MFAFragment.kt

```diff
@@ -0,0 +1,236 @@
+                    displaySignedOutState()
+                } else {
+                    displayDialog(getString(R.string.unexpected_sdk_result_title), signOutResult.toString())
+                }
+            }
+        }
+    }
+
+    private fun displaySignedInState(accountState: AccountState) {
+        emptyFields()
+        updateUI(STATUS.SignedIn)
+        displayAccount(accountState)
+    }
+
+    private fun displaySignedOutState() {
+        emptyFields()
+        updateUI(STATUS.SignedOut)
+        emptyResults()
+    }
+
+    private fun updateUI(status: STATUS) {
+        when (status) {
+            STATUS.SignedIn -> {
+                binding.signIn.isEnabled = false
+                binding.signOut.isEnabled = true
+            }
+            STATUS.SignedOut -> {
+                binding.signIn.isEnabled = true
+                binding.signOut.isEnabled = false
+            }
+        }
+    }
+
+    private fun emptyFields() {
+        binding.emailText.setText("")
+        binding.passwordText.setText("")
+    }
+
+    private fun emptyResults() {
+        binding.resultAccessToken.text = ""
+        binding.resultIdToken.text = ""
+    }
+
+    private fun displayAccount(accountState: AccountState) {
+        CoroutineScope(Dispatchers.Main).launch {
+            val accessTokenResult = accountState.getAccessToken()
+            when (accessTokenResult) {
+                is GetAccessTokenResult.Complete -> {
+                    val accessToken = accessTokenResult.resultValue.accessToken
+                    binding.resultAccessToken.text = getString(R.string.result_access_token_text) + accessToken
+
+                    val idToken = accountState.getIdToken()
+                    binding.resultIdToken.text = getString(R.string.result_id_token_text) + idToken
+                }
+                is GetAccessTokenError -> {
+                    displayDialog(getString(R.string.msal_exception_title), accessTokenResult.errorMessage)
+                }
+            }
+        }
+    }
+
+    private fun handleSignInError(error: SignInError) {
+        when {
+            error.isInvalidCredentials() || error.isBrowserRequired() || error.isUserNotFound() -> {
+                displayDialog(error.error, error.errorMessage)
+            }
+            else -> {
+                // Unexpected error
+                displayDialog(getString(R.string.unexpected_sdk_error_title), error.errorMessage)
+            }
+        }
+    }
+
+    private fun displayDialog(error: String? = null, message: String?) {
+        val builder = AlertDialog.Builder(requireContext())
+        builder.setTitle(error)
+            .setMessage(message)
+        val alertDialog = builder.create()
+        alertDialog.show()
+    }
+
+    private fun navigateToMFARequired(nextState: MFARequiredState) {
+        val bundle = Bundle()
+        bundle.putParcelable(Constants.STATE, nextState)
+        val fragment = MFACodeFragment()
```

Comment:
Let's call it "challenge" instead of "code". In the context of MFA we're moving away from "code" to the more generic "challenge"

Reply:
Done. Update all layout and fragment from code to challenge